lib/model.js

```diff
@@ -3380,17 +3372,15 @@ Model.bulkWrite = async function bulkWrite(ops, options) {
   }
   options = options || {};
 
-  const shouldSkip = await new Promise((resolve, reject) => {
-    this.hooks.execPre('bulkWrite', this, [ops, options], (err) => {
-      if (err != null) {
-        if (err instanceof Kareem.skipWrappedFunction) {
-          return resolve(err);
-        }
-        return reject(err);
+  const shouldSkip = await this.hooks.execPre('bulkWrite', this, [ops, options]).then(
+    () => null,
+    err => {
```

Comment:
```suggestion
  const shouldSkip = await this.hooks.execPre('bulkWrite', this, [ops, options]).catch(err => {
```

Small nitpick, consider using `.then(resolve, reject)` or `.catch(reject)` more consistently, or did this return need to be mapped to `null`, but the others not?